common/include/pcl/impl/point_types.hpp

```diff
@@ -1894,31 +1894,31 @@ POINT_CLOUD_REGISTER_POINT_STRUCT (pcl::_PointXYZRGBNormal,
     (float, x, x)
     (float, y, y)
     (float, z, z)
-    (float, rgb, rgb)
     (float, normal_x, normal_x)
     (float, normal_y, normal_y)
     (float, normal_z, normal_z)
+    (float, rgb, rgb)
     (float, curvature, curvature)
```

Comment:
The changes in this file were pushed to
- show, that those adaptations would solve the issues with the test cases in this PR.
- check, whether those changes would break other tests.

I'm aware that you probably don't want to have these changes in this PR as they are not really related to #5539. I will revert them / move them to a separate PR if you like. Otherwise, just tell me to keep them here.

**Why are those changes actually needed?**
The order in the sequences, which are defined in those macros need to be aligned with the order of the elements in the respective struct definitions. Otherwise `toPCLPointCloud2 ()` will fill the `fields` vector in the wrong order (means fields with a higher offset will be added before fields with lower offsets).